### ISSUE TITLE
Support for ES 6.6?

### ISSUE THREAD

**Issue by CDBridger** [IBM/elasticsearch-spark-recommender#48]
I'm extremely new to ES but I've been going through the notebook changing code where required to make it work (e.g multiple types per index are no longer supported). I'm now stuck at the retrieving/calculating the recommendations part of the example (after calculating the embeddings and bringing them back into ES).

Specifically in the fn_query this part of the ES query is out of date:

```
"script": {
"inline": "payload_vector_score",
"lang": "native",
"params": {
"field": "@model.factor",
"vector": query_vec,
"cosine" : cosine
}
}
```

fails and I get the error :
`RequestError: RequestError(400, 'search_phase_execution_exception', 'script_score: the script could not be loaded')`

And I'm not sure how to get this up to speed for ES 6.6.

Cheers!

EDIT: Ah, I'm assuming the script is a plugin written by MLnick in his other repository? Which I notice has a TODO for porting to latest ES.

**Comment by stevemart** [comment]
@MLnick poke ☝️

**Comment by Minx-SigEp** [comment]
I too would be interested in seeing this updated to a later version of elasticsearch. With the removal of types from indexes, the introduction of the [dense_vector data type](https://www.elastic.co/guide/en/elasticsearch/reference/7.6/dense-vector.html), and the [vector scoring plugin](https://github.com/MLnick/elasticsearch-vector-scoring) being incompatible with ES 5.5+ this sample could benefit from an update. There is a lot of extremely valuable information available here that is hard to apply to a current version of Elasticsearch.

**Comment by MLnick** [comment]
Hi all, apologies for not replying sooner. The new dense vector data type and embedded `cosineSimilarity` function in ES scripting language looks very exciting and useful.

I will look at updating the project to use this functionality, if possible, and undertake some performance comparisons.

**Comment by MLnick** [comment]
Hi all - see #54 for an update to 7.x

**Comment by MLnick** [comment]
#54 is merged, this repo will only support ES 7.6 +
